libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/NavigationView.kt

```diff
@@ -188,6 +195,49 @@ class NavigationView : ConstraintLayout {
         updateNavigationViewOptions(navigationViewOptions)
     }
 
+    private val configurationOnStyleLoadListener = OnStyleLoadedListener {
+        removeConfigureOnStyleLoadListener()
+        mapView.location.setLocationProvider(navigationLocationProvider)
+        mapView.location.enabled = true
+
+        MapboxDropInUtils.getLastLocation(
+            this@NavigationView.context,
+            WeakReference<(Expected<Exception, LocationEngineResult>) -> Unit> { result ->
+                result.fold(
+                    {
+                        Log.e(TAG, "Error obtaining current location", it)
+                    },
+                    {
+                        it.lastLocation?.apply {
+                            cameraViewModel.consumeLocationUpdate(this)
+                            navigationLocationProvider.changePosition(this, listOf(), null, null)
+                            initLocationComponentPuck()
+                        }
+                    }
+                )
+            }
+        )
+
+        initializedObserver?.initialized()
```

Comment:
Why should developers be concerned about knowing when the style has loaded (granted they don't interact with the Maps SDK themselves) in the first place? it feels to me like it's not necessary and we can manage this state internally in the drop-in view models.

If someone sets a route while the map style has not loaded yet, our view models should be smart enough and reduce the state to not try to update the route, and do that only after the style has loaded.

If developers do plan to interact with the Map SDK directly, they can listen to style changes themselves.

Reply:
It's not so much about the developer knowing the style is loaded it's about notifying the developer the NavigationView is finished initializing.  Nothing can really happen until the map is fully loaded but that detail is abstracted from the external developer.  However `MapboxNavigation` will be emitting various events constantly, for example on device rotation. NavigationView shouldn't take for granted it can handle these events at any time.  I would propose moving our existing observers to only subscribe to the view model flows after the map is fully loaded.  

In addition, developers can add their own observers.  If those external observers are called even though the map hasn't fully loaded it can be a bit misleading to external developer because (I think) the assumption would be the developer can confidently act on those events. 

I think we should examine this closer and only give out the API interface after we know that the map and NavigationView are fully loaded/initialized.  This would prevent an external developer from even adding observers to NavigationView until it has fully initialized.

Reply:
> Nothing can really happen until the map is fully loaded but that detail is abstracted from the external developer. [...] If those external observers are called even though the map hasn't fully loaded it can be a bit misleading to external developer because (I think) the assumption would be the developer can confidently act on those events.

I don't fully agree with this. Developers could still act on the current user location, or our components could still take advantage of progress (to update the maneuver view or play instruction) even before the style is fully loaded (for any reason). I wouldn't couple these together.

In my opinion the style being loaded should only be another input to state reducers (being worked on in https://github.com/mapbox/mapbox-navigation-android/pull/5111, cc @abhishek1508) for components that do need to have access to style (route line / camera / recenter button).

If developers fetch the raw reference of the map and interact with it themselves, they should be aware of the style availability dependency before performing any style related actions. This should be clear from Maps SDK docs.

Reply:
Ok, I'll remove it but I think it's going to be a problem in the future.